crates/codegen/grammar/src/parser_definition.rs

```diff
@@ -4,6 +4,13 @@ use std::rc::Rc;
 use crate::visitor::{GrammarVisitor, Visitable};
 use crate::{PrecedenceParserDefinitionRef, ScannerDefinitionRef, VersionQualityRange};
 
+/// A named wrapper, used to give a name to a [`ParserDefinitionNode`].
+#[derive(Clone, Debug)]
+pub struct Named<T> {
```

Comment:
do you think this would be a good candidate for `impl Deref for Named<T>` to its `node`? I see a few `X.node.Y` expressions that can be shortened here. But not a blocker of course.

Reply:
Makes sense; added in https://github.com/NomicFoundation/slang/pull/710/commits/8ee53140ce3f05aca1e574021b90e79e59cb81c7